### ISSUE TITLE
redundant_clone: false positive using ToOwned

### ISSUE THREAD

**Issue by nicomem** [rust-lang/rust-clippy#6729]
The following code :
```rust
pub fn own<T, U>(v: T) -> U
where
    T: ToOwned<Owned = U>,
{
    v.to_owned()
}
```

generates the following lint:
```
warning: redundant clone
 --> src/main.rs:5:6
  |
5 |     v.to_owned()
  |      ^^^^^^^^^^^ help: remove this
  |
  = note: `#[warn(clippy::redundant_clone)]` on by default
note: this value is dropped without further use
 --> src/main.rs:5:5
  |
5 |     v.to_owned()
  |     ^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_clone
```

However, removing the `.to_owned` leads to a compilation error since the return types are different.

May be related to #5700 

### Meta

- `cargo clippy -V`: clippy 0.1.52 (e9920ef 2021-02-11)
- `rustc -Vv`:
  ```
  rustc 1.52.0-nightly (e9920ef77 2021-02-11)
  binary: rustc
  commit-hash: e9920ef7749d11fc71cc32ca4ba055bcfeaab945
  commit-date: 2021-02-11
  host: x86_64-unknown-linux-gnu
  release: 1.52.0-nightly
  LLVM version: 11.0.1
  ```

**Comment by camsteffen** [comment]
The problem is that your function takes a value instead of a reference. But it would be better if `redundant_clone` was more clear about that. So this is an enhancement rather than a false positive.

**Comment by camsteffen** [comment]
In fact, it really should be `needless_pass_by_value` instead.

**Comment by humb1t** [comment]
@camsteffen - I had similar issue and if I change function to take a reference it would still cause complains from clippy.

**Comment by humb1t** [comment]
```rust
    9         pub fn new<P: AsRef<Path> + ToOwned<Owned = PathBuf>>(root_path: &P) -> Self {
    8             Configuration {
    7                 #[allow(clippy::redundant_clone)]
    6                 root_path: root_path.to_owned(),
    5                 stream_read_ahead: 10,
    4                 stream_buffer_size: CHUNK_BYTES_SIZE,
    3             }
    2         }
```

**Comment by pitaj** [comment]
@humb1t I wasn't able to reproduce your issue ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1afd6bb341d5ae134fbc3e9364776d08))

Perhaps it's been fixed. Can you provide a complete example?

**Comment by humb1t** [comment]
@pitaj looks like it was fixed. Thank you!

**Comment by flip1995** [comment]
Just noticed that the original example still triggers the lint.

**Comment by Kimundi** [comment]
Lint still triggers, another short example:
```rust
fn example<T: ToOwned<Owned = String>>(v: T) -> String {
    v.to_owned()
}
```

**Comment by xtqqczze** [comment]
Lint still triggers with Rust 1.93: https://godbolt.org/z/o5Ed6ocjM
